ros2_batch_job/packaging.py

```diff
@@ -56,52 +52,50 @@ def build_and_test_and_package(args, job):
         cmd.extend(cmake_args)
 
     job.run(cmd)
-
-    if ros1_bridge_ignore_marker:
-        os.remove(ros1_bridge_ignore_marker)
     print('# END SUBSECTION')
 
     # only build the bridge on Linux for now
     # the OSX nodes don't have a working ROS 1 installation atm and
     # ROS1 is not supported on Windows
     if args.os in ['linux']:
-        print('# BEGIN SUBSECTION: build ROS 1 bridge')
-        # Now run build only for the bridge
-        env = dict(os.environ)
-        env['MAKEFLAGS'] = '-j1'
-        job.run([
-            args.colcon_script, 'build',
-            '--base-paths', '"%s"' % args.sourcespace,
-            '--build-base', '"%s"' % args.buildspace,
-            '--install-base', '"%s"' % args.installspace,
-            '--cmake-args', '-DBUILD_TESTING=ON', '--no-warn-unused-cli',
-            '--packages-select', 'ros1_bridge',
-            '--event-handlers', 'console_direct+',
-        ] + (['--merge-install'] if not args.isolated else []) +
-            (
-                ['--cmake-args', '-DCMAKE_BUILD_TYPE=' +
-                    args.cmake_build_type]
-                if args.cmake_build_type else []
-        ), env=env)
-        print('# END SUBSECTION')
+        if args.mixed_ros_overlay_pkgs:
+            print('# BEGIN SUBSECTION: build overlay packages')
+
+            overlay_pkgs = ' '.join(args.mixed_ros_overlay_pkgs)
+            # Now run build only for the bridge
+            env = dict(os.environ)
+            env['MAKEFLAGS'] = '-j1'
+            job.run([
+                args.colcon_script, 'build',
+                '--base-paths', '"%s"' % args.sourcespace,
+                '--build-base', '"%s"' % args.buildspace,
+                '--install-base', '"%s"' % args.installspace,
+                '--cmake-args', '-DBUILD_TESTING=ON', '--no-warn-unused-cli',
+                '--event-handlers', 'console_direct+',
+            ] + (['--packages-select', overlay_pkgs]) +
```

Comment:
Each parameter needs to be a separate items in the list. Joining multiple with a space is wrong here. Needs to be:

`] + ['--packages-select'] + args.mixed_ros_overlay_pkgs +`

Same below.